be/src/vec/exec/scan/file_scanner.cpp

```diff
@@ -913,26 +916,32 @@ Status FileScanner::_get_next_reader() {
         const TFileRangeDesc& range = _current_range;
         _current_range_path = range.path;
 
-        if (!_partition_slot_descs.empty()) {
-            // we need get partition columns first for runtime filter partition pruning
-            RETURN_IF_ERROR(_generate_parititon_columns());
+        // try to get the partition columns from the range
+        RETURN_IF_ERROR(_generate_partition_columns());
+        // try to get the data lake partition columns from the range
+        RETURN_IF_ERROR(_generate_data_lake_partition_columns());
 
-            if (_state->query_options().enable_runtime_filter_partition_prune) {
-                // if enable_runtime_filter_partition_prune is true, we need to check whether this range can be filtered out
-                // by runtime filter partition prune
-                if (_push_down_conjuncts.size() < _conjuncts.size()) {
-                    // there are new runtime filters, need to re-init runtime filter partition pruning ctxs
-                    _init_runtime_filter_partition_prune_ctxs();
-                }
+        const auto& partition_col_descs = !_partition_col_descs.empty()
```

Comment:
I think we can merge `_partition_col_descs` and `_data_lake_partition_col_descs`.
Because we only need one of them. The only diff is that one is generated from path, one is from metadata.